Add pending review status to unapproved  reviews

rossanafmenezes opened the pull request:
Fixes #1763

## Changes

- Adds a Pending review label and change the notif Icon colour on unapproved reviews
- Forces a reloadReviewsFromCache() after the status of a review is changed to display the new status on the review list screen. 

## Screenshots

| Before | After |
| ---- | ---- |
| <img src="https://user-images.githubusercontent.com/30724184/129804077-bfcc631a-25af-434b-af88-f4f1affd6950.png" width="350"/> | <img src="https://user-images.githubusercontent.com/30724184/129804074-2725def1-8521-4d94-b858-957157bc32da.png" width="350"/> |
| <img src="https://user-images.githubusercontent.com/30724184/129804363-5a515c73-8214-4769-8f21-b2bbc81124df.png" width="350"/> | <img src="https://user-images.githubusercontent.com/30724184/129804361-91086d7c-9788-47b6-ad41-eed6c02db181.png" width="350"/> |

### Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->

1 - Go to WooCommerce - Settings - Products on a browser
2 - Enable Reviews
3 - Leave a new review on a product (while logged out)
4 - Open the app and navigate to the Reviews tab
5 - Note the Pending Review Label and the icon on a different colour
6 - click on the review and change the status
7 - Note the status updated on the review list screen (although it takes a few seconds)

### Video

https://d.pr/v/GJygiQ


- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

rossanafmenezes committed:
Add pending label to unapproved reviews

Add a Pending review label and changes the notifIcon colour on unapproved product reviews

rossanafmenezes committed:
Refresh review list screen on status changed

forces a reloadReviewsFromCache after the review status is changed and the reviews list is loaded again.

anitaa1990 commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/reviews/ReviewListAdapter.kt:
```diff
@@ -406,7 +411,31 @@ class ReviewListAdapter(
+            val pendingReviewLabel: String = viewBinding.root.context.getString(
+                R.string.pending_review_label
+            )
+            val reviewText: String = StringUtils.getRawTextFromHtml(review.review)
+            val spannableReviewText = SpannableStringBuilder(pendingReviewLabel)
+            val pendingLabelColor: Int = ContextCompat.getColor(viewBinding.root.context, R.color.woo_orange_50)
+            val notifsIconPendingColor: Int = ContextCompat.getColor(viewBinding.root.context, R.color.woo_purple_60)
+
+            spannableReviewText.setSpan(
+                ForegroundColorSpan(pendingLabelColor),
+                0, // start
+                14, // end
+                Spannable.SPAN_INCLUSIVE_INCLUSIVE
+            )
+            spannableReviewText.insert(17, reviewText)
```
I had a couple of suggestions here. Let me know what you think 😄  

1. Do you think we can move the logic for the Spannable text inside the if else block? Since it will only be used if the review status is on hold, we only need this logic if the condition is true.
2. We don't want to hardcode the 17 here 🤔 . Since all the strings from the `strings.xml` file will be translated, this number could vary depending on the translated language and it might not always be 17. We could just use the length of the `Pending review text` here. 

So something like:
```
<string name="pending_review_label">Pending Review</string>
```

Then in code:
```
spannableReviewText.insert(pendingReviewLabel.length, reviewText)
```

Another way to maybe go would be to use `HTML.fromHtml` instead of using Spannable. We do something similar in [ProductItemViewHolder](https://github.com/woocommerce/woocommerce-android/blob/45c31a1b06019e6e52aee69e32aa81edf36ec58b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductItemViewHolder.kt#L23).

I tried an example implementation [here](https://github.com/woocommerce/woocommerce-android/compare/issue/1763-add-pending-review-status...test/issue-1763?expand=1). Let me know your thoughts 😄

anitaa1990 commented on WooCommerce/src/main/res/values/strings.xml:
```diff
@@ -1916,4 +1916,5 @@
     <string name="product_downloadable_files_download_settings">Download Settings</string>
     <string name="shipping_label_payments_cant_edit_warning">Only the site owner can manage the shipping label payment methods. Please contact Store Owner %1$s (%2$s) to manage payment methods.</string>
     <string name="magic_link_not_meaning_to_create_account">Didn\'t mean to create a new account? Go back to re-enter your email address.</string>
+    <string name="pending_review_label">Pending Review • \</string>
```
Super nitpick: we tend to group strings based on the screen/feature they are used in. Commonly used strings are at the top of the file and if you look at the [review list strings](https://github.com/woocommerce/woocommerce-android/blob/f75f53f7f15a98ae10cf1a219e9ee4139a7c2ed1/WooCommerce/src/main/res/values/strings.xml#L917), they are grouped together.

anitaa1990 reviewed (commented):
Fantastic work here @rossanafmenezes 🥇 ! Thank you for taking this one! I tested the changes and it works great! 

I left some code suggestions for you review. Let me know your thoughts! There are some nitpicks in the code as well - feel free to follow your own judgment on whether you want to incorporate those 😄 

There is also a lint error in the PR. We use Detekt to check for lint issues in the code before opening a PR. For instance, from your android studio terminal, if you use `./gradlew detektAll`, it will point out the lint issues in the code so you can fix them.

rossanafmenezes committed:
update pending review label solution

Update solution to use HtmlCompat instead of spannable as per reviewe suggestions
moved string to the related group of strings in the file.

rossanafmenezes committed:
remove white spaced

white spaced removed as per lint error

ctarda committed:
Reformat code.

rossanafmenezes commented:
Hi @anitaa1990 

Thank you for reviewing it. I have made the changes you suggested. Thank you for the input.

anitaa1990 reviewed (approved):
Looks good :shipit: Thank you @rossanafmenezes!

rossanafmenezes commented:
Thank you @anitaa1990 🙏
